### ISSUE TITLE
OCP Registry PVC get overwritten 

### ISSUE THREAD

**Issue by mjudeikis** [openshift/openshift-ansible#4507]
#### Description

I had my OCP cluster running for long time. PV and PVC for the registry was configured manually as per user guide. (patch deployment).
After upgrade to 3.5 my storage got overwritten to emtyDir().

This big covers 3 potential fixes I already ran into in the few different customers. 

Long story short: Ansible for 3.5 OCP overwrites default storage backend to emptyDir() so I need option to point my registry storage to custom locations.

**3 use cases**:
1. I want the ability to specify for Ansible DO NOT TOUCH my registry deployment config.
2. I want to be able to use configuration variables to specify which PVC to use.
3. I want to be able to use nonstandard names for NFS exports 

**Case 1:**
I have supplementary groups configured and custom another config. I want Ansible update just image version when upgrading. 
`openshift_hosted_registry_preserve_dc=true`

**Case 2:**
my docker registry is backed with PVC with customer name: 
```
oc get pvc
ovc-7tfgq Bound Registry
```
I want to set:
`openshift_hosted_regsitry_pvc_name=ovc-7tfgq` so when ansible runs it makes sure this PVC is used.

**Case 3**:
I want to use nonstandart names in the ansible to create my pv and pvc:
Ansible setup:
```
openshift_registry_selector='purpose=infra'
openshift_hosted_registry_storage_kind=nfs
openshift_hosted_registry_storage_access_modes=['ReadWriteMany']
openshift_hosted_registry_storage_host=backend.bcs-nas4.prod.example.com
openshift_hosted_registry_storage_nfs_directory=/ifs/data/vol11
openshift_hosted_registry_storage_volume_name=/stg_ci5
openshift_hosted_registry_storage_volume_size=10Gi
```
Openshift will construct Mount path:
`backend.bcs-nas4.prod.example.com:/ifs/data/vol11/stg_ci5` and it will fail with error:
```
TASK [openshift_persistent_volumes : Create PersistentVolumes] *****************
fatal: [app-cimaint-p1.example.com]: FAILED! => {
    "changed": false,
    "cmd": [
        "oc",
        "create",
        "-f",
        "/tmp/openshift-ansible-IS15Ze6/persistent-volumes.yml",
        "--config=/tmp/openshift-ansible-IS15Ze6/admin.kubeconfig"
    ],
    "delta": "0:00:00.156036",
    "end": "2017-06-20 15:54:13.796851",
    "failed": true,
    "failed_when_result": true,
    "rc": 1,
    "start": "2017-06-20 15:54:13.640815",
    "warnings": []
}
 
STDERR:
 
The PersistentVolume "stg_ci5-volume" is invalid: metadata.name: Invalid value: "stg_ci5-volume": must match the regex [a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)* (e.g. 'example.com')
```
I want to be able to setup:

```
openshift_registry_selector='purpose=infra'
openshift_hosted_registry_storage_kind=nfs
openshift_hosted_registry_storage_access_modes=['ReadWriteMany']
openshift_hosted_registry_storage_host=backend.bcs-nas4.prod.example.com
openshift_hosted_registry_storage_nfs_directory=/ifs/data/vol11
openshift_hosted_registry_storage_volume_name=/stg_ci5
openshift_hosted_registry_storage_volume_size=10Gi
openshift_hosted_registry_storage_volume_pvc_name=registry
```
where last variable would be used to name PVC ans avoid clases with naming standart.

**Comment by mangirdaz** [comment]
+ to case 1:
I want to use supplementary groups for my storage backend so I added:
```
securityContext:
  supplementalGroups:
  - 12345
```
And it gets overwritten when rerunning ansible. I want to preserve this config.

**Comment by mangirdaz** [comment]
now customer always runs:
`ansible-playbooks -i invetory openshift-ansible/playbooks/byo/config.yml --skip-tags hosted`

**Comment by gregswift** [comment]
It might be worth noting that things such as endpoints are not removed/overwritten during subsequent runs.  This is mainly relevant to gluster backed PVCs but has been a constant thorn in our side

**Comment by mangirdaz** [comment]
Yes. But I think it will be fixed in 3.6

**Comment by mjudeikis** [comment]
solved.
